packages/mdc-data-table/README.md

```diff
@@ -1,4 +1,258 @@
+`mdc-data-table-row-text-color($color)` | Sets the row text color.
+`mdc-data-table-shape-radius($radius)` | Sets the rounded shape with given radius size. `$radius` can be single radius or list radius values up to 4 list size.
+`mdc-data-table-stroke-size($size)` | Sets the border size of data-table.
+`mdc-data-table-stroke-color($color)` | Sets the border color of data-table.
+`mdc-data-table-header-row-height($height)` | Sets the header row height.
+`mdc-data-table-row-height($height)` | Sets row height.
+`mdc-data-table-cell-padding($leading-padding, $trailing-padding)` | Sets leading & trailing padding for all cells.
+`mdc-data-table-column-widths($width-list)` | Sets the custom widths for each table column.
+
+## Accessibility
+
+Please refer [WAI-ARIA Authoring Practices for table](https://www.w3.org/TR/wai-aria-practices-1.1/#table) for ARIA recommended role, states & properties required for table element.
+
+## Events
+
+Please use MDCDataTable's constants file to access these event constants.
+
+```ts
+const {events} from '@material/data-table/constants';
+// `events.ROW_SELECTION_CHANGED` to access event constant.
+```
+
+Event constant | Event name | Description
+-- | -- | --
+`ROW_SELECTION_CHANGED` | `MDCDataTable:changed` | Event emitted when row checkbox is checked or unchecked.
+`SELECTED_ALL` | `MDCDataTable:selectedAll` | Event emitted when header row checkbox is checked.
+`UNSELECTED_ALL` | `MDCDataTable:unselectedAll` | Event emitted when header row checkbox is unchecked.
+
+## `MDCDataTable` Properties and Methods
+
+Method Signature | Description
+--- | ---
+`layout() => void` | Registers new row checkboxes, header row checkbox and updates the state of header row checkbox. Use this when rows are added / removed from data table.
+`getRows() => HTMLElement[]` | Returns array of row elements.
+`getSelectedRowIds() => Array<string|nul>` | Returns array of selected row ids.
+`setSelectedRowIds(rowIds: string[])` | Sets selected row ids. Overwrites previously selected rows.
+
+## Usage within Web Frameworks
+
+If you are using a JavaScript framework, such as React or Angular, you can create a Data Table for your framework. Depending on your needs, you can use the _Simple Approach: Wrapping MDC Web Vanilla Components_, or the _Advanced Approach: Using Foundations and Adapters_. Please follow the instructions [here](../../docs/integrating-into-frameworks.md).
+
+### `MDCDataTableAdapter`
+
+Method Signature | Description
+--- | ---
+`addClassAtRowIndex(rowIndex: number, cssClasses: string) => void` | Adds a class name to row element at given row index excluding header row.
+`getRowCount() => number` | Returns row count excluding header row.
+`getRowElements() => HTMLElement[]` | Returns array of row elements excluding header row.
+`getRowIdAtIndex(rowIndex: number) => string | null` | Returns row id of row element at given row index based on `data-row-id` attribute on row element `tr`.
```

Comment:
this is rendered incorrectly

Reply:
Good catch. Add escape character for pipe symbol. Updated this in PR 4891